src/tree.rs

```diff
@@ -135,3 +149,125 @@ pub fn minimum_spanning_tree(
 
     Ok(spanning_tree)
 }
+
+/// Bipartition tree by finding balanced cut edges of a spanning tree using
+/// node contraction. Assumes that the tree is connected and is a spanning tree.
+/// A balanced edge is defined as an edge that, when cut, will split the
+/// population of the tree into two connected subtrees that have population near
+/// the population target within some epsilon. The function returns a list of
+/// all such possible cuts, represented as the set of nodes in one
+/// partition/subtree.
```

Comment:
Did you have any citations here or other links to the use or source of this algorithm you'd want to add to the documentation?

Reply:
This balanced cut edge finding algo is not actually specified in the literature, I believe (the paper I linked to uses a naive `O(e^2)` algo that calculates the population balance for each possible edge and throws away the unsuitable cut edges, whereas the one I implemented is `O(v+e)` and does BFS traversal). I can link to the original Python implementation, though, if you'd like.

Reply:
No that's fine. I was thinking it would be nice additional context for the documentation if it was available. But it's not critical and I think the description here is probably sufficient. Maybe having an example to show how it's used and what the expected result looks like would be nice. But not worth blocking over either.